runtime/reconcilers/model.go

```diff
@@ -1521,3 +1515,31 @@ func md5Hash(val []byte) (string, error) {
 	}
 	return hex.EncodeToString(hash.Sum(nil)), nil
 }
+
+// shouldTriggerReset determines if a model should trigger a reset based on its change mode and the current state compared to the specified hash.
+func (r *ModelReconciler) shouldTriggerReset(model *runtimev1.Model, specHash string, exists bool) bool {
+	// These conditions trigger a reset regardless of change mode
+	if model.Spec.TriggerFull ||
+		model.State.ResultConnector == "" ||
+		model.State.RefreshedOn == nil ||
+		!exists {
+		return true
+	}
+
+	// Handle different change modes
+	switch model.Spec.ChangeMode {
+	case "manual":
+		// In manual mode, spec hash changes don't trigger automatic resets
+		// Only explicit trigger or missing output will cause a reset
+		return false
+	case "patch":
+		// In patch mode, silently update the hash but don't trigger a reset
+		if model.State.SpecHash != specHash {
+			model.State.SpecHash = specHash
+		}
+		return false
```

Comment:
1. It should error if the `patch` change mode is used with a non-incremental model.
2. Let's emit an info log that the definition was changed and that we're changing to it without a full refresh. 
3. It needs to call `r.C.UpdateState` for changes to `model.State` to take effect